implementations/micrometer-registry-stackdriver/src/test/java/io/micrometer/stackdriver/StackdriverIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2025 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.stackdriver;
+
+import io.micrometer.common.util.internal.logging.InternalLogLevel;
+import io.micrometer.common.util.internal.logging.MockLogger;
+import io.micrometer.common.util.internal.logging.MockLoggerFactory;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Timer;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * When running this integration test locally, authentication can be done by configuring
+ * the env var GOOGLE_APPLICATION_CREDENTIALS with the location of the service key JSON
+ * file.
+ */
+@Tag("gcp-it")
+class StackdriverIntegrationTest {
+
+    static final MockLoggerFactory LOGGER_FACTORY = new MockLoggerFactory();
+
+    MockLogger mockLogger = LOGGER_FACTORY.getLogger(StackdriverMeterRegistry.class);
+
+    static final StackdriverConfig CONFIG = new StackdriverConfig() {
+        @Override
+        public String projectId() {
+            return "micrometer-gcp";
+        }
+
+        @Override
+        public String get(String key) {
+            return null;
+        }
+    };
+
+    StackdriverMeterRegistry registry = LOGGER_FACTORY
+        .injectLogger(() -> new StackdriverMeterRegistry(CONFIG, Clock.SYSTEM));
+
+    StackdriverMeterRegistry.Batch batch = registry.new Batch();
+
+    void publishAndVerify() {
+        registry.close();
+        assertThat(mockLogger.getLogEvents()
+            .stream()
+            .filter(log -> log.getLevel() == InternalLogLevel.WARN || log.getLevel() == InternalLogLevel.ERROR))
+            .as("There should be no WARN or ERROR logs if metrics publish to Stackdriver successfully")
+            .isEmpty();
+    }
+
+    @Test
+    void timerWithSlos() {
+        Timer slos = Timer.builder("micrometer.stackdriver.it.timer.slo")
+            .maximumExpectedValue(Duration.ofMillis(2))
+            .serviceLevelObjectives(Duration.ofMillis(1), Duration.ofMillis(2))
+            .register(registry);
+        slos.record(Duration.ofMillis(1));
+        slos.record(Duration.ofMillis(2));
+        slos.record(Duration.ofMillis(3));
+
+        System.out.println(slos.getId().toString() + batch.distribution(slos.takeSnapshot(), true));
```

Comment:
Should we keep the `System.out.println` statements?

Reply:
I think they're nice for debugging in the IDE when there are issues. They won't show up in the Gradle output generally, so it seems low cost to leave them. Thoughts?